File: Algorithms/src/961. N-Repeated Element in Size 2N Array/Code/solution.go
func repeatedNTimes(nums []int) int {
    seen := make(map[int]bool)

    for _, x := range nums {
        // If already seen, this is the repeated element
        if seen[x] {
            return x
        }
        // Otherwise, mark as seen
        seen[x] = true
    }
    return -1 // Guaranteed answer exists
}

File: Algorithms/src/961. N-Repeated Element in Size 2N Array/Code/solution.cpp
class Solution
{
public:
    int repeatedNTimes(vector<int> &nums)
    {
        unordered_set<int> seen;

        for (int x : nums)
        {
            // If already seen, this is the repeated element
            if (seen.count(x))
            {
                return x;
            }
            // Otherwise, mark it as seen
            seen.insert(x);
        }
        return -1; // Will never reach here due to problem guarantee
    }
};
